sklearn/externals/_lobpcg.py

```diff
@@ -19,7 +19,7 @@
 .. [3] A. V. Knyazev's C and MATLAB implementations:
        https://github.com/lobpcg/blopex
 """
-
+import inspect
```

Comment:
I would like to apply the following sync before making this change here:

https://github.com/scikit-learn/scikit-learn/pull/25176